cmake/compiler/host-gcc/target_nostdinc.cmake

```diff
@@ -0,0 +1,8 @@
+
+macro(toolchain_cc_nostdinc)
+
+  zephyr_compile_options(
+    -nostdinc
+  )
+
+endmacro()
```

Comment:
If we believe this `host-gcc/target_nostdinc.cmake` file will be the same as `gcc/target_nostdinc.cmake`, then I'd vote for removing this file and let host-gcc piggy back off main gcc.